nixos/modules/security/audit.nix

```
@@ -110,23 +80,35 @@ in
     };
   };
 
-  config = {
-    systemd.services.audit = {
-      description = "Kernel Auditing";
-      wantedBy = [ "basic.target" ];
+  config = lib.mkIf (cfg.enable == "lock" || cfg.enable) {
+    systemd.services.audit-rules = {
+      description = "Load Audit Rules";
+      wantedBy = [ "sysinit.target" ];
+      before = [
+        "sysinit.target"
+        "shutdown.target"
+      ];
+      conflicts = [ "shutdown.target" ];
 
       unitConfig = {
+        DefaultDependencies = false;
         ConditionVirtualization = "!container";
-        ConditionSecurity = [ "audit" ];
+        ConditionKernelCommandLine = [
+          "!audit=0"
+          "!audit=off"
+        ];
       };
 
-      path = [ pkgs.audit ];
-
       serviceConfig = {
         Type = "oneshot";
         RemainAfterExit = true;
-        ExecStart = "@${if enabled then startScript else disableScript} audit-start";
-        ExecStop = "@${stopScript} audit-stop";
+        ExecStart = "${pkgs.audit}/bin/auditctl -R ${rules}/audit.rules";
```

Nit:
```suggestion
        ExecStart = "${lib.getExe' pkgs.audit "auditctl"} -R ${rules}/audit.rules";
```
I expect a `getExe'` would be more robust in terms of outputs? Probably not much of a difference, it seems this string conversion correctly infers the use of the `$bin` output, but this kind of string shenanigans always makes me a little uneasy.